src/hardware/dac.rs

```diff
@@ -69,6 +69,40 @@ use hal::dma::{
 static mut DAC_BUF: [[[u16; SAMPLE_BUFFER_SIZE]; 3]; 2] =
     [[[0; SAMPLE_BUFFER_SIZE]; 3]; 2];
 
+/// Custom type for referencing DAC output codes.
+/// The internal integer is the raw code written to the DAC output register.
+#[derive(Copy, Clone)]
+pub struct DacCode(pub u16);
+
+impl Into<f32> for DacCode {
+    fn into(self) -> f32 {
+        // The output voltage is generated by the DAC with an output range of +/- 4.096 V. This
+        // signal then passes through a 2.5x gain stage. Note that the DAC operates using unsigned
+        // integers, and u16::MAX / 2 is considered zero voltage output. Thus, the dynamic range of
+        // the output stage is +/- 10.24 V. At a DAC code of zero, there is an output of -10.24 V,
+        // and at a max DAC code, there is an output of (slightly less than) 10.24 V.
+
+        let dac_volts_per_lsb = 4.096 * 2.5 / (1u16 << 15) as f32;
+
+        // Note that the bipolar table is an offset-binary code, but it is much more logical and
+        // correct to treat it as a twos-complement value. TO do that, we XOR the most significant
+        // bit to convert it.
+        (self.0 ^ 0x8000) as i16 as f32 * dac_volts_per_lsb
+    }
```

Comment:
There's some issues with placing tests in the `stabilizer` crate - specifically, Rust can't find the crate `test`. As such, I'm avoiding tests, but I think it would be value-added in the future to add some small sanity check tests (e.g. verify code of 0 equates to near our negative scale, code of 0xFFFF equates to near our positive scale).